src/app/plan/duck/sagas.ts

```diff
@@ -693,6 +719,274 @@ function* migrationPoll(action) {
   }
 }
 
+/******************************************************************** */
+//Hooks sagas
+/******************************************************************** */
+
+function* fetchHooksSaga(action) {
+  const state = yield select();
+  try {
+    const { migMeta } = state;
+    const client: IClusterClient = ClientFactory.cluster(state);
+    const migHookResource = new MigResource(MigResourceKind.MigHook, migMeta.namespace);
+    const { currentPlan } = state.plan
+    const getPlanRes = yield call(getPlanSaga, currentPlan.metadata.name);
+    const currentPlanHooks = getPlanRes.data.spec.hooks;
+
+    const hookList = yield client.list(migHookResource);
+    const associatedHooks = []
+    if (currentPlanHooks) {
+      currentPlanHooks.forEach((currentPlanHookRef) => hookList.data.items.forEach((hookRef) => {
+        if (currentPlanHookRef.reference.name === hookRef.metadata.name) {
+          const hookImageType = hookRef.spec.custom ? 'custom' : 'ansible'
+          const customContainerImage = hookRef.spec.custom ? hookRef.spec.image : null;
+          const ansibleRuntimeImage = !hookRef.spec.custom ? hookRef.spec.image : null;
+          const srcServiceAccountName = hookRef.spec.targetCluster === 'source' ? currentPlanHookRef.serviceAccount : null;
+          const srcServiceAccountNamespace = hookRef.spec.targetCluster === 'source' ? currentPlanHookRef.executionNamespace : null;
+          const destServiceAccountName = hookRef.spec.targetCluster === 'destination' ? currentPlanHookRef.serviceAccount : null;
+          const destServiceAccountNamespace = hookRef.spec.targetCluster === 'destination' ? currentPlanHookRef.executionNamespace : null;
+          const clusterTypeText = hookRef.spec.targetCluster === 'destination' ? 'Target cluster' : 'Source cluster';
+
+          let ansibleFile;
+          if (!hookRef.spec.custom) {
+            ansibleFile = atob(hookRef.spec.playbook);
+          }
+
+          const uiHookObject = {
```

Comment:
I think this would be another usage of that HookDefinition type interface

Reply:
This one is tricky because I had to combine some values from MigPlan & MigHook to satisfy the ui table requirements. I am in favor of coming back and hitting this one as part of the ts support second pass.